### DISCUSSION TITLE
Visual Shader - make it blender-like

### DISCUSSION THREAD

**Discussion by Cr0wned** [godotengine/godot-proposals#4029] [Editor]
Proposal for making the visual shader editor more usable, with blender-like nodes (mapping, procedural textures) to make complex materials easily (that way)
![Capture d’écran de 2022-02-19 16-19-03](https://user-images.githubusercontent.com/67225326/154806997-6eafeb55-a29e-476d-8b65-01ec31848b7f.png)

**Comment by Cr0wned** [comment]
Btw there is material maker but I mean, something similar directly integrated right in the engine would be more practical for both people not knowing about material maker and those who don't wanna download extra softwares :)

**Reply by ultramarinebicycle** [reply]
Other engines tend to use a bridge plugin when all the functionality exists in another software, which is especially great when the software is made using the engine itself! The engine developers are very mindful of the binary size and duplicating functionality is likely to be a no-go!

**Comment by eon-s** [comment]
I think that this could be implemented via plugin with a set of `VisualShaderNodeCustom` scripts.

The general problem of plugin-based solutions is the discoverability of these plugins, the visual shader window could have a button that points to the asset library with the corresponding filters.
